ftdc/cmd/parser.go

```diff
@@ -144,10 +190,81 @@ func main() {
 		panic(err)
 	}
 
-	gpw := newGnuPlotWriter()
-	for _, flatDatum := range data {
-		gpw.addFlatDatum(flatDatum)
-	}
+	stdinReader := bufio.NewReader(os.Stdin)
+	render := true
+	graphOptions := defaultGraphOptions()
+	for {
+		if render {
+			gpw := newGnuPlotWriter(graphOptions)
+			for _, flatDatum := range data {
+				gpw.addFlatDatum(flatDatum)
+			}
 
-	gpw.RenderAndClose()
+			gpw.Render()
+		}
+		render = true
+
+		// This is a CLI. It's acceptable to output to stdout.
+		//nolint:forbidigo
+		fmt.Print("$ ")
+		cmd, err := stdinReader.ReadString('\n')
+		switch {
+		case err != nil && errors.Is(err, io.EOF):
+			nolintPrintln("\nExiting...")
+			return
+		case cmd == "quit":
+			nolintPrintln("Exiting...")
+			return
+		case cmd == "h" || cmd == "help":
+			render = false
+			nolintPrintln("range <start> <end>")
+			nolintPrintln("-  Only plot datapoints within the given range. \"zoom in\"")
+			nolintPrintln("-  E.g: range 2024-09-24T18:00:00 2024-09-24T18:30:00")
+			nolintPrintln("-  All times in UTC")
+			nolintPrintln()
+			nolintPrintln("reset range")
+			nolintPrintln("-  Unset any prior range. \"zoom out to full\"")
+			nolintPrintln()
+			nolintPrintln("`quit` or Ctrl-d to exit")
+		case strings.HasPrefix(cmd, "range "):
+			pieces := strings.SplitN(cmd, " ", 3)
+			// TrimSpace to remove the newline.
+			start, end := pieces[1], strings.TrimSpace(pieces[2])
+			_, _ = start, end
+
+			if start == "start" {
+				graphOptions.minTimeSeconds = 0
+			} else {
+				goTime, err := time.Parse("2006-01-02T15:04:05", start)
```

Comment:
Are we expecting CLI users to enter start and end times in this format? May want to include that format in the error message on parse failure.

Reply:
FWIW, the format was included in the `help`. I'll add it here too.